tidy3d/web/api/container.py

```diff
@@ -462,13 +477,17 @@ class Batch(WebContainer):
         description="Collection of parent task ids for each job in batch, used internally only.",
     )
 
-    jobs: Dict[TaskName, Job] = pd.Field(
-        None,
-        title="Simulations",
-        description="Mapping of task names to individual Job object for each task in the batch. "
-        "Set by ``Batch.upload``, leave as None.",
+    num_workers: Optional[pd.PositiveInt] = pd.Field(
+        DEFAULT_NUM_WORKERS,
+        title="Number of Workers",
+        description="Number of workers for multi-threading upload and download of batch. "
+        "Corresponds to ``max_workers`` argument passed to "
+        "``concurrent.futures.ThreadPoolExecutor``. When left ``None``, will pass the maximum "
+        "number of threads available on the system.",
```

Comment:
Hmm I was reading about this as it was interesting:

https://docs.python.org/3/library/concurrent.futures.html#threadpoolexecutor
> Changed in version 3.8: Default value of max_workers is changed to min(32, os.cpu_count() + 4). This default value preserves at least 5 workers for I/O bound tasks. It utilizes at most 32 CPU cores for CPU bound tasks which release the GIL. And it avoids using very large resources implicitly on many-core machines. ThreadPoolExecutor now reuses idle worker threads before starting max_workers worker threads too.

Sounds like the default when it is None is limited to avoid it consuming all the resources and maybe it changed since they wrote the original docs, so maybe we want to reference this as maybe this affects the super power users who want to max out their CPUs

Reply:
hm, for me `os.cpu_count() + 4` is 12. 🤔